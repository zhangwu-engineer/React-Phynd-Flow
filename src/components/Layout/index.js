import React from 'react'
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { a11yProps } from 'utils/helper';

const useStyles = makeStyles(theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  toolbar: theme.mixins.toolbar,
  moduleBar: {
    boxShadow: 'none'
  }
}));

const moduleRouter = [
  '/provider-module/provider-details',
  '/location-module/location-details',
  '/network-module/network-details',
  '/healthplan-module/healthplan-details'
]

const LayoutContainer = ({ children, history }) => {
  const classes = useStyles();
  const [activeTab, setActiveTab] = React.useState(0);

  const handleChange = (event, newTab) => {
    setActiveTab(newTab);
    history.push(moduleRouter[newTab]);
  };

  return (
    <div>
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <Typography variant="h6" noWrap>Channel Builder</Typography>
        </Toolbar>
        <Paper square className={classes.moduleBar}>
          <Tabs
            indicatorColor="primary"
            textColor="primary"
            variant="scrollable"
            scrollButtons="auto"
            aria-label="scrollable auto tabs example"
            value={activeTab}
            onChange={handleChange}
          >
            <Tab label="Provider Module" {...a11yProps(0)} />
            <Tab label="Location Module" {...a11yProps(1)} />
            <Tab label="Network Module" {...a11yProps(2)} />
            <Tab label="HealthPlan Module" {...a11yProps(3)} />
          </Tabs>
        </Paper>
      </AppBar>
      {children}
    </div>
  )
}

export default LayoutContainer
