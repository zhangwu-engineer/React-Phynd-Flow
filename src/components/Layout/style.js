import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  toolbar: theme.mixins.toolbar,
  moduleBar: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    boxShadow: 'none',
  },
  buttonGroup: {
    boxShadow: 'none',
  }
}));