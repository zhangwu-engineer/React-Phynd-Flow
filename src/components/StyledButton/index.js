import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

export default withStyles({
  root: {
    borderRadius: 0,
    border: 0,
    color: 'white',
    height: 34,
    padding: '0 30px',
    marginRight: 24,
  },
  label: {
    textTransform: 'capitalize',
    fontSize: 14,
  },
})(Button);