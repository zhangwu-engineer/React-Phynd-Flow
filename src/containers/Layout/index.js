import React from 'react'

class LayoutContainer extends React.Component {
  render() {
    return(
      <div>
        {this.props.children}
      </div>
    )
  }
}

export default LayoutContainer
