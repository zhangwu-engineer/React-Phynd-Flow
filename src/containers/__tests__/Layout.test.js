import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Layout from 'components/Layout';
import StyledButton from 'components/StyledButton';

let layoutComponent;
const historyMock = { push: jest.fn() };
const submitCTA = jest.fn();
const revertCTA = jest.fn();

describe('To test the Layout Component functionality.', () => {
  configure({ adapter: new Adapter() });

  beforeEach(() => {
    layoutComponent = shallow(
      <Layout
        history={historyMock}
        submitCTA={submitCTA}
        revertCTA={revertCTA}
      />
    );
  });

  it("renders without crashing", () => {
    const layoutComponentDefault = shallow(<Layout />);
    expect(layoutComponentDefault).toMatchSnapshot();
  });

  it("renders with history props", () => {
    expect(layoutComponent).toMatchSnapshot();
  });

  it("generates 3 StyledButtons", () => {
    expect(layoutComponent.find(StyledButton)).toHaveLength(3);    
  });

  it("simulate SubmitAll button trigger", () => {
    const submitAllButton = layoutComponent.find(StyledButton).first();
    submitAllButton.simulate('click');
    expect(submitCTA.mock.calls.length).toEqual(1);
  });

  it("simulate RevertAll button trigger", () => {
    const revertAllButton = layoutComponent.find(StyledButton).at(2);
    revertAllButton.simulate('click');
    expect(revertCTA.mock.calls.length).toEqual(1);
  });


});
