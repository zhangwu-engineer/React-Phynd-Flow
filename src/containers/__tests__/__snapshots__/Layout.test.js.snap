// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the Layout Component functionality. renders with history props 1`] = `ShallowWrapper {}`;

exports[`To test the Layout Component functionality. renders without crashing 1`] = `ShallowWrapper {}`;
